mk/rustllvm.mk

```diff
@@ -43,6 +43,10 @@ $$(RT_OUTPUT_DIR_$(1))/$$(call CFG_STATIC_LIB_NAME_$(1),rustllvm): \
 	@$$(call E, link: $$@)
 	$$(Q)$$(call CFG_CREATE_ARCHIVE_$(1),$$@) $$^
 
+EXTRA_RUSTLLVM_CXXFLAGS_$(1) = $$(shell echo $$(filter $$(LLVM_ALL_COMPONENTS_$(1)),\
+	$$(LLVM_SUBTARGET_COMPONENTS)) | tr 'a-z' 'A-Z' |\
+	sed -e 's/^ //;s/\([^ ]*\)/\-DLLVM_SUBTARGET_\1/g')
```

Comment:
Hmm... it does not work, because the `-` character is not allowed in a define, but it is used in several components. Something like this seems to work:
```
RUSTLLVM_COMPONENTS_$(1) = $$(shell echo $$(LLVM_ALL_COMPONENTS_$(1)) |\
        tr 'a-z-' 'A-Z_'| sed -e 's/^ //;s/\([^ ]*\)/\-DLLVM_COMPONENT_\1/g')
```
and still avoids the distinction between target and non-target components.